### ISSUE TITLE
osx compile error: No timer function available

### ISSUE THREAD

**Issue by chuckcranor** [pmodels/argobots#13]
argobots currently does not compile on osx due to an #error in cond.c

I looked at the code, and the timer code in cond.c seems to duplicate the timer code in abtd_time.c except that the code in cond.c does not handle the ABT_CONFIG_USE_MACH_ABSOLUTE_TIME case (thus the error).

Seems like the thing to do is use the ABTD_time_read_sec() API in cond.c and get rid of the machine dependent ifdefs in it.

```
> make argobots
[  0%] Performing build step for 'argobots'
Making all in src
  CC       libabt_la-cond.lo
/Users/chuck/excluded/pdlfs/extern-umbrella/build/argobots-prefix/src/argobots/src/cond.c:134:2: error: 
      "No timer function available"
#error "No timer function available"
 ^
1 error generated.
make[5]: *** [libabt_la-cond.lo] Error 1
make[4]: *** [all-recursive] Error 1
make[3]: *** [argobots-prefix/src/argobots-stamp/argobots-build] Error 2
make[2]: *** [CMakeFiles/argobots.dir/all] Error 2
make[1]: *** [CMakeFiles/argobots.dir/rule] Error 2
make: *** [argobots] Error 2
```

**Comment by smseo** [comment]
Thanks for reporting this. I've fixed this issue and pushed the patch (c017b4994b6e3c7dfc285bd6501c98cf7b2f5625). Please try the master again, and if you encounter the same issue, please let me know.

**Comment by chuckcranor** [comment]
The patched worked!  thanks!
